### PR TITLE
Update docker file template for core

### LINKED ISSUE #685: Update Docker template for RC2
There is no AFAIK updated Docker image for use with dotnet/aspnet under RC2.
The current `microsoft/dotnet` is old and content from RC2 branch won't work on this image:
```
dotnet --version
.NET Command Line Tools (1.0.0-beta-001598)

Product Information:
 Version:     1.0.0-beta-001598
 Commit Sha:  N/A

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  14.04
 OS Platform: Linux
 Runtime Id:  ubuntu.14.04-x64
```
The template needs to ship with SQLite path (as the current one) in order to allow runing starter web projects out-of-the-box.

### PR DESCRIPTION
**Pull request by spboyer**
Fixes #685.

/cc
@OmniSharp/generator-aspnet-team-push

### CODE REVIEW

File: templates/Dockerfile.txt [hunk 1]
```diff
@@ -1,11 +1,12 @@
-<% if(!coreclr){ %>FROM microsoft/aspnet:1.0.0-rc1-update1<% } %><% if(coreclr){ %>FROM microsoft/aspnet:1.0.0-rc1-update1-coreclr<% } %>
+<% if(!coreclr){ %>FROM microsoft/dotnet<% } %><% if(coreclr){ %>FROM microsoft/dotnet:core<% } %>
 
-RUN printf "deb http://ftp.us.debian.org/debian jessie main\n" >> /etc/apt/sources.list
-RUN apt-get -qq update && apt-get install -qqy sqlite3 libsqlite3-dev && rm -rf /var/lib/apt/lists/*
+#Set urls and port environments
+ENV ASPNETCORE_URLS="http://*:5000
```
Comment (akamud) [inline templates/Dockerfile.txt]: I believe you are missing a `"` after `5000`

**Comment by glennc** [comment]
microsoft/dotnet and microsoft/dotnet:core are both coreclr based. The difference is that one only has core dependencies installed, the other has the full SDK. There is no full .NET Docker base that we ship now. There is a preview one for Windows, but no Linux one. For a Mono app (which is what !coreclr would be on Linux) we probably should look at the existing Mono image. But I haven't actually run an app on it yet.

**Comment by peterblazejewicz** [comment]
@glennc Thanks!
@spboyer See my PR: https://github.com/OmniSharp/generator-aspnet/pull/715/files
I've unified subgenerator according to new CoreCRL
Thanks!

File: templates/Dockerfile.txt [hunk 2]
```diff
@@ -1,11 +1,12 @@
-<% if(!coreclr){ %>FROM microsoft/aspnet:1.0.0-rc1-update1<% } %><% if(coreclr){ %>FROM microsoft/aspnet:1.0.0-rc1-update1-coreclr<% } %>
+<% if(!coreclr){ %>FROM microsoft/dotnet<% } %><% if(coreclr){ %>FROM microsoft/dotnet:core<% } %>
 
-RUN printf "deb http://ftp.us.debian.org/debian jessie main\n" >> /etc/apt/sources.list
-RUN apt-get -qq update && apt-get install -qqy sqlite3 libsqlite3-dev && rm -rf /var/lib/apt/lists/*
+#Set urls and port environments
+ENV ASPNETCORE_URLS="http://*:5000"
```
Comment (DavidRouyer) [inline templates/Dockerfile.txt]: ENV ASPNETCORE_SERVER.URLS="http://*:5000" instead of ENV ASPNETCORE_URLS="http://*:5000" ?

Comment (peterblazejewicz) [inline templates/Dockerfile.txt]: @DavidRouyer 
I think the variable is correct, they key is "ASPNETCORE_{VALUE}":
https://github.com/aspnet/Hosting/blob/dev/src/Microsoft.AspNetCore.Hosting.Abstractions/WebHostDefaults.cs
The ASPNETCORE_URLS is not documented, but its full support is coming: https://github.com/aspnet/Hosting/pull/760/commits/e7b8c3f90a781a55658b4245571c5bf5dac5e56e

Comment (DavidRouyer) [inline templates/Dockerfile.txt]: @peterblazejewicz Alright, this is post-RC2 stuff then! For RC2, "ASPNETCORE_URLS" doesn't work.

Comment (peterblazejewicz) [inline templates/Dockerfile.txt]: Technically no templates are able to use ASPNETCORE_URLS AFAIK. I've hit that problem previously:
> The -p option is supported in dotnet-cli but --server-urls support needs to be determined
https://github.com/OmniSharp/generator-aspnet/pull/715
The solution is to modify web application startup either by:

- explicit use of `UseUrls`: https://github.com/aspnet/Templates/issues/574#issuecomment-220727859
- explicit use of 'hosting.json`: https://github.com/peterblazejewicz/recipes/blob/master/src/gulp-dotnet-browser-sync/StarterWeb/Program.cs#L12-L13

The first blog post about RC2 + Docker also has to use that implementation: http://laurentkempe.com/2016/05/16/ASP-NET-Core-RC2-Docker-and-HipChat-Connect-add-on/

Comment (DavidRouyer) [inline templates/Dockerfile.txt]: Current templates are missing the configuration part with environment variables.
For RC2, I added :
`var config = new ConfigurationBuilder().AddEnvironmentVariables("ASPNETCORE_").Build();` and `.UseConfiguration(config)` in Program.cs, and `ENV ASPNETCORE_SERVER.URLS="http://*:5000"` in my Dockerfile and it works just fine! Looks like I'll have to use `ENV ASPNETCORE_URLS="http://*:5000"` post-RC2.

**Comment by peterblazejewicz** [comment]
@spboyer Shayne, closing in favour of #743 
Thanks!

**Pull request closed without merging**